src/Orleans.Core.Abstractions/IDs/ClientGrainId.cs

```diff
@@ -1,11 +1,13 @@
 using System;
 using System.Collections.Generic;
+using Orleans.Concurrency;
 
 namespace Orleans.Runtime
 {
     /// <summary>
     /// Identifies a client.
     /// </summary>
+    [Immutable]
     internal readonly struct ClientGrainId : IEquatable<ClientGrainId>, IComparable<ClientGrainId>
```

Comment:
These should never be transmitted, and hence shouldn't need the `[Immutable]` attribute

Reply:
But they might still get deep-copied by the SerializationManager? I just noticed that they get code generated for deep-copier/serializer/deserializer. Maybe it would be a good idea to add a NonSerializableAttribute (or something similar so that SerializationManager refuses to process these types (or at least codegeneration)?